graphql-mode.el

```diff
@@ -281,15 +281,22 @@ Please install it and try again."))
 (defun graphql-indent-line ()
   "Indent GraphQL schema language."
   (let ((position (point))
+        (column)
         (indent-pos))
+    (save-excursion
```

Comment:
Nice! 👍 

I wonder if we could simplify this a little bit.

Having a look at the state returned by `syntax-ppss`:

> The list of the positions of the currently open parentheses, starting with the outermost.

could we use this to avoid the loop?  so `goto-char` the outermost parenthesis?

It would also be kind of nice to do a single `syntax-ppss` call if possible.

Reply:
Another food finding: `syntax-ppss-toplevel-pos state`:

> This function extracts, from parser state state, the last position scanned in the parse which was at top level in grammatical structure. “At top level” means outside of any parentheses, comments, or strings.

> The value is nil if state represents a parse which has arrived at a top level position.

Reply:
Ok, good find on `syntax-ppss-toplevel-pos state`. That helped eliminate the loop. I also rewrote `graphql-beginning-of-query` so that I can use it in `graphql-indent-line`. If you don't like this change, I can revert it.

Reply:
I think it looks great! 👍